run.sh

```diff
@@ -1,19 +1,34 @@
-if [ -f "${USER_NAME_SECRET}" ]; then
-    read USR < ${USER_NAME_SECRET}    
-    COMMAND_OPTIONS="${COMMAND_OPTIONS} -username $USR"
+#!/usr/bin/env sh
+${JENKINS_USERNAME_FILE:=/run/secrets/jenkins-swarm-agent}
+if [ -f "${JENKINS_USERNAME_FILE}" ]; then  
+    echo "Read Jenkins username form file ${JENKINS_USERNAME_FILE}"
+    read JENKINS_USERNAME < ${JENKINS_USERNAME_FILE}  
 fi
 
-if [ -f "${PASSWORD_SECRET}" ]; then
-    read PSS < ${PASSWORD_SECRET}
-    export PSS
-    COMMAND_OPTIONS="${COMMAND_OPTIONS} -passwordEnvVariable PSS"
+if [ -n "${JENKINS_USERNAME_PASSWORD}" ]; then
+    echo "Using Password from the Env Variable JENKINS_USERNAME_PASSWORD"
+    PASWD_OPT="-passwordEnvVariable ${JENKINS_USERNAME_PASSWORD}"  
 fi
 
+${JENKINS_USERNAME_PASSWORD_FILE:=/run/secrets/jenkins-swarm-agent-password}
+if [ -f "${JENKINS_USERNAME_PASSWORD_FILE}" ]; then
+    echo "Read Jenkins username password form file ${JENKINS_USERNAME_PASSWORD_FILE}"
+    PASWD_OPT="-passwordFile ${JENKINS_USERNAME_PASSWORD_FILE}"
+fi
+
+if [ -z "${JENKINS_URL}" ]; then
+    echo "Warning: The JENKINS_URL is not set, Using default URL http://jenkins:8080"
+    JENKINS_URL="http://jenkins:8080"
+fi
+
+# Always try to download swarm-client from current jenkins.
+wget ${JENKINS_URL}/swarm/swarm-client.jar 
+
 # GROUP=$(stat -c %G /var/run/docker.sock)
 # if [ ! $(grep ^$GROUP: /etc/group) ]; then
 #     addgroup -G $(stat -c %g /var/run/docker.sock) $GROUP
 # fi
 # adduser jenkins $GROUP
 
-# exec su jenkins -c "exec java -jar /home/jenkins/swarm-client-${SWARM_CLIENT_VERSION}.jar ${COMMAND_OPTIONS}"
-java -jar /home/jenkins/swarm-client-${SWARM_CLIENT_VERSION}.jar ${COMMAND_OPTIONS}
\ No newline at end of file
+# exec su jenkins -c "exec java -jar /home/jenkins/swarm-client.jar ${COMMAND_OPTIONS}"
+java -jar /swarm-client.jar -master ${JENKINS_URL:="http://jenkins:8080"} -username ${USR:="jenkins-swarm-agent"} ${PASWD_OPT} ${SWARM_CLIENT_OPTS}
```

Comment:
Maybe we should add an `if` statement so that `COMMAND_OPTIONS` is used if it has a value, otherwise, the new command? That way we'd maintain backward compatibility.  What do you think?

Reply:
Do you mean something like this:

```sh
if env.${COMMAND_OPTIONS}.isNotEmpty() then
java -jar /swarm-client.jar ${COMMAND_OPTIONS}
else
java -jar /swarm-client.jar -master ${JENKINS_URL:="http://jenkins:8080"} -username ${USR:="jenkins-swarm-agent"} ${PASWD_OPT} ${SWARM_CLIENT_OPTS}
```